Cd and Ls commands were implemented

faucct opened the pull request:
Неудобства, которые я заметил:

* Команды, работающие с файлами, читают целиком в буфер их содержимое. Это создаст проблемы с гигантскими файлами.
* Пустой `README`. Из-за этого и того, что в тестах список команд пайплайна назывался `lines`, 
я сначала пытался дописать результат `ls` в конец `prevResult`.
* Управление ошибками с помощью `IOException`, который никак не обрабатывается, хотя может быть брошен не из кода CLI и значить что-то важное.

Плюсы:

* Команды было просто реализовать.
* Есть тесты.
* Тесты было просто написать.

faucct committed:
Cd and Ls commands were implemented

artbez commented on src/main/java/ru/ifmo/sdc/cli/commands/CdCommand.java:
```diff
@@ -0,0 +1,35 @@
+        final Path relativePath = Paths.get(params.isEmpty() ? environment.get("HOME") : params.get(0));
+        final Path path = relativePath.isAbsolute() ?
+                relativePath :
+                Paths.get(environment.get("PWD"), relativePath.toString()).toAbsolutePath();
```
Где в первый раз кладется PWD в environment? 
Если предполагается, что из System.getenv(), то у меня там он не лежит, например
Отсюда простой вызов ```ls``` порождает Execution interrupted

faucct replied:
Готово.

artbez commented on src/main/java/ru/ifmo/sdc/cli/commands/LsCommand.java:
```diff
@@ -0,0 +1,47 @@
+        if (file.exists())
+            if (file.isDirectory())
+                for (String subpath : file.list())
+                    stringBuilder.append(subpath).append('\n');
```
Тут хорошо бы либо без последнего "\n", либо убрать заменить println на print в момент печати строки. Иначе получается лишняя пустая строчка

faucct replied:
Готово.

artbez commented on src/main/java/ru/ifmo/sdc/cli/commands/CdCommand.java:
```diff
@@ -0,0 +1,35 @@
+        }
+        environment.put(
+                "PWD",
+                path.toString()
```
path.normalize().toString()
Иначе могут возникать такие вот штуки
```cd: /path/to/project/software-design-course/../test: No such file or directory```

faucct replied:
Готово.

artbez commented on src/main/java/ru/ifmo/sdc/cli/commands/CdCommand.java:
```diff
@@ -0,0 +1,35 @@
+ * Change working directory to first argument or home.
+ */
+public class CdCommand extends Command {
+    @CommandLine.Parameters(index = "1..*")
```
Еще небольшое замечание. Было бы здорово, чтобы в случае нескольких параметров, переданных команде  cd, отображалась ошибка

faucct replied:
Готово.

faucct committed:
PR fixes
Environment::userDir
LsCommand output should have no additional endline
CdCommand should normalize userDir
CdCommand should assert number of arguments

artbez commented on src/main/java/ru/ifmo/sdc/cli/commands/CdCommand.java:
```diff
@@ -0,0 +1,32 @@
+ * Change working directory to first argument or home.
+ */
+public class CdCommand extends Command {
+    @CommandLine.Parameters(index = "1..1")
```
Похоже, что все-таки так проблема не решатся и нужно честно проверять size
```
cd .. .. 
$ null$ 
```

faucct replied:
Поправил вывод:

```
$ cd .. ..
Unmatched argument: ..
```

faucct committed:
Shell ParameterException output
